cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -453,29 +311,91 @@ angular.module(PKG.name + '.commons')
       };
     }
 
-    function formatConnections() {
-      var connections = [];
-      angular.forEach(vm.instance.getConnections(), function (conn) {
-        connections.push({
-          from: conn.sourceId,
-          to: conn.targetId
-        });
+    function addConnection(newConnObj) {
+      $scope.connections.push({
+        from: newConnObj.sourceId,
+        to: newConnObj.targetId
+      });
+      DAGPlusPlusNodesActionsFactory.setConnections($scope.connections);
+    }
+
+    function removeConnection(detachedConnObj, updateStore = true) {
+      var detachedConnection = _.find($scope.connections, function (conn) {
+        return conn.from === detachedConnObj.sourceId && conn.to === detachedConnObj.targetId;
+      });
+      if (detachedConnection) {
+        $scope.connections.splice($scope.connections.indexOf(detachedConnection), 1);
+      }
+      if (updateStore) {
```

Comment:
Yeah, in `removeSelectedConnections`, we're manually calling this function for each selected connection that we detach, but we don't want it to update the store every time, since it would mess up the undo & redo. Only after we have removed all the selected connections do we update the store.